src/coreapplication.cpp

```diff
@@ -163,7 +163,7 @@ void CoreApplication::showMainWindow()
     disconnect(&_notification, &Notification::messageClicked, this,
                &CoreApplication::showMainWindow);
 
-    _mainWindow = new MainWindow();
+    _mainWindow = new MainWindow(&_taskManager);
```

Comment:
Passing TaskManager to MainWindow like that breaks the entire encapsulation and loose binding design in that:
1. TaskManager doesn't need to know/care about what's on top of it;
2. MainWindow doesn't have to be tightly coupled with TaskManager in any way;
The only information passing method right now between the UI and backend code is strictly Qt signals&slots mechanism. This implies that the connection between the UI and backend components is loose, non-binding, self encapsulated and segregated. If one was to extend his imagination, the current cross-platform backend is easily portable to any other type of interface where Qt is available (mobile, embedded or headless daemon), which was part of the original design. I think it's best that we refrain from tightly coupling backend code with UI if there's no outstanding argument against that.